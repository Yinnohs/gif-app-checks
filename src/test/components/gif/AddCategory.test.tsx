import { describe, it, expect, vi } from "vitest";
import { render, screen, fireEvent } from "@testing-library/react";
import { AddCategory } from "../../../components";
describe("testing Add category component", () => {
  it("should be the same as the snapshot", () => {
    const { container } = render(<AddCategory onNewCategory={() => {}} />);
    expect(container).toMatchSnapshot();
  });

  it("should change the input value", () => {
    const changedValue = "GOKU";
    render(<AddCategory onNewCategory={() => {}} />);

    const input = screen.getByRole("textbox");

    fireEvent.input(input, { target: { value: changedValue } });

    expect(input.getAttribute("value")).toEqual(changedValue);
  });

  it("should change input value when fired button event", () => {
    const inputValue = "GOKU";
    render(<AddCategory onNewCategory={() => {}} />);

    const input = screen.getByRole("textbox");
    const button = screen.getByRole("button");

    fireEvent.input(input, { target: { value: inputValue } });
    fireEvent.click(button);

    expect(input.getAttribute("value")?.length).toBe(0);
  });

  it("should change be called handler method with the input value", () => {
    const inputValue = "GOKU";
    const mockedFunction = vi.fn();

    render(<AddCategory onNewCategory={mockedFunction} />);
    const input = screen.getByRole("textbox");
    const button = screen.getByRole("button");

    fireEvent.input(input, { target: { value: inputValue } });
    fireEvent.click(button);

    expect(mockedFunction).toHaveBeenCalledTimes(1);
    expect(mockedFunction).toHaveBeenCalledWith(inputValue);
  });

  it("should not  calle handler method with the input value in blank", () => {
    const inputValue = "";
    const mockedFunction = vi.fn();

    render(<AddCategory onNewCategory={mockedFunction} />);
    const input = screen.getByRole("textbox");
    const button = screen.getByRole("button");

    fireEvent.input(input, { target: { value: inputValue } });
    fireEvent.click(button);

    expect(mockedFunction).toHaveBeenCalledTimes(0);
  });
});
