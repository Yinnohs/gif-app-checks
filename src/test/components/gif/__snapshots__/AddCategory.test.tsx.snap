// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`testing Add category component > should be the same as the snapshot 1`] = `
<div>
  <div>
    <label
      for="category-search-input"
    >
       Search 
    </label>
    <input
      id="category-search-input"
      name="category-search-input"
      placeholder="search a gif"
      type="text"
      value=""
    />
    <button
      class="btn"
      type="button"
    >
      <span>
        Add a Gif
      </span>
    </button>
  </div>
</div>
`;
